openedx/features/pakx/lms/overrides/utils.py

```diff
@@ -346,6 +346,47 @@ def create_dummy_request(site, user):
     return request
 
 
+def get_progress_statistics_by_block_types(request, course_key):
+    course_key = CourseKey.from_string(course_key)
```

Comment:
Since we've same functionality in `get_course_progress_percentage`, refactor common code in a new method.

Reply:
alright